src/NServiceBus.Gateway.PersistenceTests/When_supporting_distributed_transactions.cs

```diff
@@ -0,0 +1,112 @@
+﻿namespace NServiceBus.Gateway.PersistenceTests
+{
+    using System;
+    using System.Threading.Tasks;
+    using System.Transactions;
+    using Extensibility;
+    using NUnit.Framework;
+
+    class When_supporting_distributed_transactions : NServiceBusGatewayDeduplicationStorageTest
+    {
+        [SetUp]
+        public void IgnoreNonDTCStorages()
+        {
+            if (!storage.SupportsDistributedTransactions)
+            {
+                Assert.Ignore();
+            }
+        }
+
+        [Test]
+        public async Task Only_marks_messages_as_dispatched_when_committed()
+        {
+            var messageId = Guid.NewGuid().ToString("D");
+
+            using (var scope = new TransactionScope(TransactionScopeOption.RequiresNew, TransactionScopeAsyncFlowOption.Enabled))
+            {
+                using (var session = await storage.CheckForDuplicate(messageId, new ContextBag()))
+                {
+                    Assert.IsFalse(session.IsDuplicate);
```

Comment:
I'm not such a huge fan of mixing arrange and asserts. I can so though how this can work here. Another option would be to capture flags and then Assert at the end which would be my preference but I'm not hung up on it

Reply:
I have no strong preference either.

Reply:
I like KISS and less is more so I would choose the current form.